### PR TITLE
Implemented Food Per Agent Ratio

### LINKED ISSUE #175: Scale maxPlatFood/FoodOnPlatform according to the number of agents in the tower.
In the spec it says that everyone should be able to survive (given the right strategies), but this isn't possible as the "FoodOnPlatform" is hard coded as 100 not proportional to number of agents

### PR DESCRIPTION
**Pull request by VFjr**
Closes #175 

Added `UseFoodPerAgentRatio` and `FoodPerAgentRatio` config parameters to allow to automatically set the amount of food depending on the number of agents. 

If `UseFoodPerAgentRatio` is true, then it multiplies `FoodPerAgentRatio` by number of agents to calculate total food. It ignores the `FoodOnPlatform` parameter if it is set

If  `UseFoodPerAgentRatio` is false, then it uses `FoodOnPlatform` and ignores `FoodPerAgentRatio`

I updated the config.json to include default parameters for them, and added error handling in case any of these three parameters are not set correctly.

### CODE REVIEW

File: pkg/config/configparameters.go [hunk 1]
```diff
@@ -13,35 +13,37 @@ import (
...
-	LogFileName    string        `json:"LogFileName"`
-	NumOfAgents    []int
-	NumberOfFloors int
-	TicksPerDay    int
-	DayInfo        *day.DayInfo
+	FoodOnPlatform       food.FoodType `json:"FoodOnPlatform"`
+	FoodPerAgentRatio    int           `json:"FoodPerAgentRatio"`
+	UseFoodPerAgentRatio bool          `json:"UseFoodPerAgentRatio"`
+	Team1AgtOne          int           `json:"Team1AgtOne"`
+	Team1AgtTwo          int           `json:"Team2AgtTwo"`
```
Comment (JaafarRammal) [inline pkg/config/configparameters.go:20]: shouldn't that be `json:"Team1AgtTwo"`

**Review by JaafarRammal — DISMISSED** [review]
Thanks. Would be great to have a wiki page for the config!

**Review by r15hil — DISMISSED** [review]
LGTM thanks!

File: pkg/config/configparameters.go (hunk 1 above)
Comment (VFjr) [reply pkg/config/configparameters.go]: From a quick look this seems to be left from Team2's commit from PR #148. I saw that you also fixed it in your latest commit, and I incorporated those changes in my rebase.

**Comment by VFjr** [comment]
> Thanks. Would be great to have a wiki page for the config!

Done, we now have a Config wiki page

**Review by jzzheng22 — APPROVED** [review]

**Pull request merged**